tests/dao/authorDAO.test.js

```diff
@@ -1,65 +1,42 @@
-import Author from "../../models/Author.js"
-import AuthorDAO from "../../dao/Author.js"
-import Connection from "../../config/Connection.js"
+import Author from "../../src/models/Author.js"
+import AuthorDAO from "../../src/dao/Author.js"
+import Connection from "../../src/config/Connection.js"
 
 describe("Teste para clase Autor", () => {
-  it("Add deve lançar erro quando autor for undefided", async () => {
-    const conn = new Connection()
-    const authorDAO = new AuthorDAO(conn)
-    await expect(async () => {
-      const author = undefined
-      await authorDAO.add(author)
-    }).rejects.toThrowError(`O objeto não é do tipo autor`)
-    await conn.close()
-  })
-  it("Add deve lançar erro quando autor for null", async () => {
-    const conn = new Connection()
-    const authorDAO = new AuthorDAO(conn)
-    await expect(async () => {
-      const author = null
-      await authorDAO.add(author)
-    }).rejects.toThrowError(`O objeto não é do tipo autor`)
-    await conn.close()
-  })
+  let conn = new Connection()
+  
+  beforeAll(async () => {
+    await conn.conect()
+});
 
-  it("Add deve lançar erro quando houver outro autor igual já cadastrado", async () => {
-    const conn = new Connection()
-    const authorDAO = new AuthorDAO(conn)
-    await expect(async () => {
+afterAll( async () => {
+    await conn.close();
+})
+    it("Criando autor com sucesso", async () => {
+      const authorDAO = new AuthorDAO(conn)
+      const author = new Author("Joao", "joao@gmail.com")
+      await authorDAO.add(author)
+      expect(author.email).toBeDefined()
+    })
```

Comment:
Então, aqui a mensagem "criando autor com sucesso" não diz exatamente o que o código está fazendo.. O teste é para se certificar que o autor pode ser salvo com sucesso ou que a criação do objeto autor não lança erros? 

Veja que seu expect não certifica nada além do undefined, se for uma string vazia ou algo diferente de undefined, seu teste passa, mesmo quando não deveria, muita brecha pra falso positivo.